pkg/tbtc/deposit_sweep.go

```diff
@@ -4,15 +4,227 @@ import (
 	"crypto/ecdsa"
 	"fmt"
 	"github.com/keep-network/keep-core/pkg/bitcoin"
+	"github.com/keep-network/keep-core/pkg/chain"
+	"go.uber.org/zap"
 )
 
 const (
 	// depositSweepProposalConfirmationBlocks determines the block length of the
 	// confirmation period that is preserved after a deposit sweep proposal
 	// submission.
 	depositSweepProposalConfirmationBlocks = 20
+	// depositSweepRequiredFundingTxConfirmations determines the minimum
+	// number of confirmations that are needed for a deposit funding transaction
+	// in order to consider it a valid part of the deposit sweep proposal.
+	depositSweepRequiredFundingTxConfirmations = 6
 )
 
+// depositSweepAction is a deposit sweep walletAction.
+type depositSweepAction struct {
+	chain           Chain
+	btcChain        bitcoin.Chain
+	sweepingWallet  wallet
+	signingExecutor *signingExecutor
+	proposal        *DepositSweepProposal
+}
+
+func newDepositSweepAction(
+	chain Chain,
+	btcChain bitcoin.Chain,
+	sweepingWallet wallet,
+	signingExecutor *signingExecutor,
+	proposal *DepositSweepProposal,
+) *depositSweepAction {
+	return &depositSweepAction{
+		chain:           chain,
+		btcChain:        btcChain,
+		sweepingWallet:  sweepingWallet,
+		signingExecutor: signingExecutor,
+		proposal:        proposal,
+	}
+}
+
+// TODO: Cover this function with unit tests once everything is completed.
```

Comment:
I would consider splitting `execute()` into several smaller functions and testing them separately. Doing one big test for this whole flow will be... complicated 😬 Especially since we have multiple corner cases.

Reply:
Yep definitely. However, I plan to do so once I have this function completed so expect that in the next PR.